src/Raven.Server/Documents/DocumentFlags.cs

```diff
@@ -9,24 +9,17 @@ public enum DocumentFlags
 
         Artificial = 0x1,
         Versioned = 0x2,
-        Reserved1 = 0x4,
-        Reserved2 = 0x8,
+        Revision = 0x4,
+        Reserved = 0x8,
 
         FromIndex = 0x10,
-        FromVersionStorage = 0x20,
+        Reserved1 = 0x20,
         FromReplication = 0x40,
-        Reserved3 = 0x80,
+        FromSmuggler = 0x80,
 
-        HasAttachments = 0x100
-    }
-
-    [Flags]
-    public enum NonPersistentDocumentFlags
```

Comment:
I also think that knowing where the document came form down the line is good. But the user may be confused to see FromSmuggler or LegacyVersioned in Flags. But IMHO he'll get used to it and this is just a flags, we can put there whatever we want to decorate the document.